engine/table/src/main/java/io/deephaven/engine/table/impl/MultiJoinModifiedSlotTracker.java

```diff
@@ -25,11 +25,14 @@ public class MultiJoinModifiedSlotTracker {
      * The location that we must write to in modified slots; also if we have a pointer that falls outside the range [0,
      * pointer); then we know it is invalid
      */
-    private int pointer;
-    /** Tow many slots we have allocated */
+    private long pointer;
+    /** How many slots we have allocated */
     private int allocated;
```

Comment:
Check your `flagSource.ensureCapacity` calls. I think there's risk of integer overflow. You need to cast one of the inputs to `long`.